toonz/sources/toonz/ocaio.cpp

```diff
@@ -534,4 +535,448 @@ void ExportOCACommand::execute() {
+      if (formatIdx >= 0) {
+        lp->options()   = prefs.levelFormat(formatIdx).m_options;
+        formatSpecified = true;
+      }
+    }
+    if (formatSpecified) {
+      sl->getProperties()->setWhiteTransp(lp->whiteTransp());
+      sl->getProperties()->setDoPremultiply(lp->doPremultiply());
+      sl->getProperties()->setDoAntialias(lp->antialiasSoftness());
+      sl->getProperties()->setDpi(lp->getDpi());
+      sl->getProperties()->setDpiPolicy(lp->getDpiPolicy());
+      // sl->getProperties()->setImageDpi(lp->getImageDpi()); //?
+      sl->getProperties()->setSubsampling(lp->getSubsampling());
+      sl->getProperties()->setColorSpaceGamma(lp->colorSpaceGamma());
+    } else {
+      sl->getProperties()->setWhiteTransp(m_whiteTransp);
+      sl->getProperties()->setDoPremultiply(m_doPremultiply);
+      sl->getProperties()->setDoAntialias(m_antialiasSoftness);
+      // we filmmakers don't use dpi ??
+      sl->getProperties()->setDpi(m_dpi);
+      sl->getProperties()->setDpiPolicy(LevelProperties::DP_CustomDpi);
+    }
+
+    // float colorSpaceGamma   = 2.2;
+    // we don't need to touch this...
+    // display gamma 2.2 is only valid for srgb color space, and for rec709 is
+    // grading on a computer monitor...but... gamma for rec709 should be 2.4 if
+    // you have a rec709 grading monitor set for HDTV, gamma for rec2020 should
+    // be 2.6 if you have a rec2020 grading monitor, gamma for DCI-P3 should
+    // be 2.4 if you have a DCI-P3 grading theatre !? linear images colorDepth
+    // could be could be F16 or F32 (but the channel
+    // width preference is locked to 32 bit if checking linear color space...) 
+    //if (sl->getPath().getType() == "exr")
+    //    sl->getProperties()->setColorSpaceGamma(colorSpaceGamma); 
+    // Why is colorSpaceGamma disabled in the level settings ui ??
+    // see render>output settings> color settings
+
+    sl->setName(jsonLayer["name"].toString().toStdWString());
+    std::vector<TFrameId> fids;
+    for (auto frame : jsonLayer["frames"].toArray()) {
+      importOcaFrame(frame.toObject(), sl);
+      TFrameId fid(frame.toObject()["frameNumber"].toInt());
+      fids.push_back(fid);
+    }
+    
+    // Shouldn't api provide a simple method to expose the level using fids in xsheet timing ?
+    int emptyColumnIndex = m_xsheet->getFirstFreeColumnIndex();
+    int frameOffset = 0;
+    if (m_startTime < 1) { // prevent crash if startTime is less than 1 : offset the timing...
+      frameOffset = -m_startTime + 1;
+    }
+    for (auto fid : fids) {
+      if (fid.getNumber() <= 0) {
+        DVGui::warning(
+            QObject::tr("importOcaLayer : skipping frame -1! ") +
+            jsonLayer["name"].toString());
+        continue;
+      }
+      // -1 converts framenumber to index, frameOffset moves the frame range above 0...
+      m_xsheet->setCell(fid.getNumber() - 1 + frameOffset, emptyColumnIndex,
+                        TXshCell(sl, fid));
+    }
```

Comment:
the `jsonLayer["duration"]` should be applied here, only if duration > 1 and  `!Preferences::instance()->isImplicitHoldEnabled()` (explicit holds mode).  Essentially copy the same `TXshCell(sl, fid)` into the next few cells based on the duration value.